ui/widgets/datepicker.js

```diff
@@ -1703,32 +1725,104 @@ $.extend( Datepicker.prototype, {
 			this._daylightSavingAdjust( new Date( drawYear, drawMonth - stepMonths, 1 ) ),
 			this._getFormatConfig( inst ) ) );
 
-		prev = ( this._canAdjustMonth( inst, -1, drawYear, drawMonth ) ?
-			"<a class='ui-datepicker-prev ui-corner-all' data-handler='prev' data-event='click'" +
-			" title='" + prevText + "'><span class='ui-icon ui-icon-circle-triangle-" + ( isRTL ? "e" : "w" ) + "'>" + prevText + "</span></a>" :
-			( hideIfNoPrevNext ? "" : "<a class='ui-datepicker-prev ui-corner-all ui-state-disabled' title='" + prevText + "'><span class='ui-icon ui-icon-circle-triangle-" + ( isRTL ? "e" : "w" ) + "'>" + prevText + "</span></a>" ) );
+		if ( this._canAdjustMonth( inst, -1, drawYear, drawMonth ) ) {
+			prev = $( "<a></a>" )
```

Comment:
The string I used is natively compliant with both HTML & XHTML, although for the single tag case it doesn't matter much as jQuery uses a shortcut that just does `document.createElement` then. I can change it to `"<a>"`.

Reply:
Using `<a>` would be more consistent with what we have in other (more modern, UI widget factory using) widgets.

Reply:
@fnagel Although it's safe to do a single unclosed tag, an unclosed tag with attributes or several unclosed tags in a string [can be an issue](https://blog.jquery.com/2020/04/10/jquery-3-5-0-released/). It's been hard to explain these subtle differences to people so closing tags is a good practice.

I guess another option would be `<a />` which jQuery also shortcuts to `document.createElement`? Not sure if that's better or not.

Reply:
@dmethvin Sadly, the linked news does not provide a recommendation what is the best practice. Looks like best thing to do is using closed tags, right?

I would suggest to keep this consistent with all other widgets (using single, unclosed tags if no attributes are used) and then do a PR changing this to single closed tags.

Reply:
The code deciding on whether to use the shortcut or not is here:
https://github.com/jquery/jquery/blob/a70274632dc19ff4a64d7bb7657a2cc647ff38b9/src/core/parseHTML.js#L35-L43
If the regex matches, the simple route is chosen. As Dave mention, just adding an attribute will break it, though, which makes `$("<div>")` work fine in XHTML mode but `$("<div a='b'>")` would break. It's safer to always close tags for this reason. `$("<div a='b' />")` would work, though - in HTML mode the slash is ignored and in XHTML mode all tags can be self-closing. Because `/>` is generally just `>` in HTML mode, though, the fact that it works is a bit magical. But maybe that's fine.

I'm fine with making the change here as advised and modifying all single tags separately in a subsequent PR.